File: DSA/Stuti_Jain_590012149/Day_13_Question_2.py
s=input("Enter a string: ")
values = {}

for c in s:
    if c in values:
        values[c]+=1
    else:
        values[c]=1

odd=0
for v in values.values():
    if v%2!= 0:
        odd+=1

if odd<=1:
    print("True")
else:
    print("False")

File: DSA/khushichaturvedi_590012789/Day_13_question_2.c
#include <stdio.h>
#include <ctype.h>

int main() {
    char str[1000];
    int freq[256] = {0}, odd = 0;

    printf("Enter a string: ");
    fgets(str, sizeof(str), stdin);

    for (int i = 0; str[i]; i++) {
        if (isalpha(str[i]))
            freq[tolower(str[i])]++;
    }

    for (int i = 0; i < 256; i++) {
        if (freq[i] % 2 != 0)
            odd++;
    }

    if (odd <= 1)
        printf("Yes, a palindrome permutation is possible.\n");
    else
        printf("No, not possible.\n");

    return 0;
}
